### ISSUE TITLE
babel-preset-env: minimum supported target underestimates supported features

### ISSUE THREAD

**Issue by connorjclark** [babel/babel#10224]
## Bug Report

**Current Behavior**

Declaring `ios: "10.3"` as the minimum supported target applies plugins for features that are supported in ios 10.3 (example: transform-async-to-generator)

**Input Code**

Please see my repro here: https://github.com/connorjclark/babel-esmodule-preset/tree/826e1c83faaa8a62f473179a08f467e575825c3e/src-ios-10-3

Included is a README containing the babel output.

**Expected behavior/code**
A clear and concise description of what you expected to happen (or code).

**Babel Configuration (.babelrc, package.json, cli command)**

```js
{
  "presets": [
    [
      "@babel/preset-env",
      {
        "targets": {
          "ios": "10.3",
        },
        "useBuiltIns": "entry",
        "debug": true
      }
    ]
  ]
}
```

**Environment**
- Babel version(s): v7.5.4
- Node/npm version: 10.15.3
- OS: OSX
- How you are using Babel: cli

**Additional context/Screenshots**

caniuse shows that async is supported in ios 10.3: https://github.com/Fyrd/caniuse/blob/bed4faadcea2ec7719c5f43167c0e2dace892be7/features-json/async-functions.json#L285

yet this transform is applied, due to matching ios 10.3 (from debug output):
```
transform-async-to-generator { "ios":"10.3" }
```

**Comment by paulirish** [comment]
Looked into this some more.

The `transform-async-to-generator` is chosen due to https://github.com/babel/babel/blob/master/packages/babel-preset-env/data/plugins.json#L240-L250, which states that Safari 11 is the first version to fully implement async functions.  

The Saf 11 datapoint (and everything in plugins.json) comes from the kangax compat-table project, rather than caniuse. (Having two diff sources of truth in this plugin certainly lead to some confusion ;)

So why do compat-table and caniuse disagree?. It wasn't always Saf 11. https://github.com/kangax/compat-table/pull/1420 was the big finding that async arrow functions dont work in in saf 10.3. 

I've opened a PR to get the caniuse data to match: https://github.com/Fyrd/caniuse/pull/5002

cc @jridgewell since he's been involved.

**Comment by jridgewell** [comment]
Yup, this is to fix buggy behaviors. The two that I know of are the async arrow functions, and tagged template literals that aren't properly cached.

**Comment by connorjclark** [comment]
Thanks @paulirish. I was looking at the wrong data. For the data babel-preset-env has, this logic checks out.

When the caniuse data is updated, would be nice to update the data here too.

> Having two diff sources of truth in this plugin certainly lead to some confusion

yeeeea. is this unavoidable? don't the two datasets have the same purpose?

**Comment by nicolo-ribaudo** [comment]
I'm closing this since it was related to a buggy browser behavior and is "fixed" by the new `bugfixes` option:
https://babeljs.io/repl/#?browsers=ios%2010.3&build=&builtIns=false&spec=false&loose=false&code_lz=IYZwngdgxgBAZgV2gFwJYHsIxM4AnZACgEoYBvAKBhmAHdhVkYAFPdAW1RAFMA6PbiHQAbAG7cSAbgoBfChRz4ixSUA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=true&fileSize=false&timeTravel=false&sourceType=script&lineWrap=false&presets=env%2Cenv&prettier=true&targets=&version=7.9.0&externalPlugins=%40babel%2Fplugin-syntax-class-properties%407.8.3

**Comment by connorjclark** [comment]
very nice, great work!

(fixed by #11083)
